crates/subspace-networking/src/node_runner.rs

```diff
@@ -112,6 +119,53 @@ impl NodeRunner {
                 },
                 _ = self.networking_parameters_registry.run().fuse() => {
                     trace!("Network parameters registry runner exited.")
+                },
+                _ = &mut self.peer_dialing_timeout => {
+                    self.handle_peer_dialing().await;
+
+                    self.peer_dialing_timeout =
+                        Box::pin(tokio::time::sleep(Duration::from_secs(3)).fuse());
+                },
+            }
+        }
+    }
+
+    async fn handle_peer_dialing(&mut self) {
+        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
+        let local_peer_id = *self.swarm.local_peer_id();
+
+        if connected_peers.len() < CONNECTED_PEERS_THRESHOLD {
+            debug!(
+                %local_peer_id,
+                "Initiate connection to known peers [connected peers={}]", connected_peers.len()
```

Comment:
You could have use structured logs instead of trying to format it manually, especially since you have `local_peer_id` above already.